feat(content): human culture conversation about Easter (#11347)
endless-sky/endless-sky: data/human/culture conversations.txt

			label backtoship
			`	With the atmosphere steeped in nostalgia for a distant city on a distant world long past, you return to your ship.`

mission "Easter Celebrations"
	minor
	source
		planet "Harmony" "New Tortuga"
	to offer
		# Easter Sunday is on a Sunday.
		"days since epoch" % 7 == 0
		# Easter Sunday occurs between 1 to 7 days inclusive after the last paschal full moon.
		# Days are counted from the end of the year to ensure consistency with leap years.
		365 - "days until year end" > ( year % 19 * 19 + ( 15 - ( 13 + year / 100 * 8 ) / 25 + year / 100 - year / 400 ) % 30 ) % 30 + 80
		365 - "days until year end" <= ( year % 19 * 19 + ( 15 - ( 13 + year / 100 * 8 ) / 25 + year / 100 - year / 400 ) % 30 ) % 30 + 80 + 7
	on offer
		conversation
			`During a walk through the spaceport, you observe a small church tucked between a grain silo and a welding shop. Written on the door in large letters are the words "Stop by for a celebration of the Risen God!"`
			choice
				`	(Step in.)`
				`	(Move on.)`
					decline

> 			`	Inside the building, a group is singing with deafening loudness. On the stage at the front of the room, a man strums a guitar; behind him a lady softly plays the piano. A large screen displays lyrics to the current song, which you note is "Christ Our Lord has Risen Again."`

Reviewer: Hmm... would FL definitely know this song? Or would it be better to just say

```suggestion
			`	Inside the building, a group is singing with deafening loudness. On the stage at the front of the room, a man strums a guitar; behind him a lady softly plays the piano. A large screen displays lyrics to the current song, "Christ Our Lord has Risen Again."`
```

and suggest that the title is on the screen?

			choice
				`	(Head out.)`
					goto headout
				`	(Stay and enjoy the music.)`

> 			`	The pastor loops through several other hymns, eventually ending with an announcement of thanks to those who attended the celebration. As the group breaks up, you notice a bulletin board documenting several church events - a potluck on Wednesday, a notification of an outing on Saturday night, and several other items hidden by someone else.`

Reviewer: Hidden by someone? How?

Author: ```suggestion
			`	The pastor loops through several other hymns, eventually ending with an announcement of thanks to those who attended the celebration. As the group breaks up, you notice a bulletin board documenting several church events - a potluck on Wednesday, a notification of an outing on Saturday night, and several other items blocked by a passing pedestrian.`
```

			label backtoship
			`	With the atmosphere steeped in nostalgia for a distant city on a distant world long past, you return to your ship.`

mission "Set Easter Date"
	landing
	repeat
	non-blocking
	to offer
		"easter: a" != "year" % 19
	on offer
		# Uses the New Scientist version of the Meeus/Jones/Butcher algorithm to calculate the date of Easter in the Gregorian Calendar.
		"easter: a" = "year" % 19
		"easter: b" = ( ( ( "year" % 100 ) - "year" ) / -100 )
		"easter: c" = "year" % 100
		"easter: d" = ( ( ( "easter: b" % 4 ) - "easter: b" ) / -4 )
		"easter: e" = "easter: b" % 4

		"easter: g" = ( ( ( ( "easter: b" * 8 + 13 ) % 25 ) - ( "easter: b" * 8 + 13 ) ) / -25 )
		"easter: h" = ( ( 19 * "easter: a" + "easter: b" - "easter: d" - "easter: g" + 15 ) % 30 )
		"easter: i" = ( ( ( "easter: c" % 4 ) - "easter: c" ) / -4 )

		"easter: k" = "easter: c" % 4
		"easter: l" = ( ( 32 + 2 * "easter: e" + 2 * "easter: i" - "easter: h" - "easter: k" ) % 7 )
		"easter: m" = ( ( ( ( "easter: a" + 11 * "easter: h" + 19 * "easter: l" ) % 433 ) - ( "easter: a" + 11 * "easter: h" + 19 * "easter: l" ) ) / -433 )

> 		"easter: n" = ( ( ( ( "easter: h" + "easter: l" - 7 * "easter: m" + 90 ) % 25 ) - ( "easter: h" + "easter: l" - 7 * "easter: m" + 90 ) ) / -25 )

Reviewer: Couldn't this be flipped to prevent having to have negative values all over the place?
(Same for a lot of the other conditions)
```suggestion
		"easter: n" = ( ( ( "easter: h" + "easter: l" - 7 * "easter: m" + 90 ) - ( ( "easter: h" + "easter: l" - 7 * "easter: m" + 90 ) % 25 ) ) / 25 )
```

Author: I asked bene about this one, and is there a good reason to not have it be negative? Since these numbers aren't player-facing, having the numbers be positive doesn't really matter.

Reviewer: I guess not... is this the usual way the formulas are presented? If so, then it's probably good to keep it.

Otherwise, it would be better to not have the negatives because that makes it less concise and clear.

Author: I think bene took it straight from the website, so yeah, this would be the way others do it.

		"easter: p" = ( ( "easter: h" + "easter: l" - 7 * "easter: m" + 33 * "easter: n" + 19 ) % 32 )

		"easter month" = "easter: n"
		"easter day" = "easter: p"
		fail

mission "Easter Celebrations"
	minor
	source
		planet "Harmony" "New Tortuga"
	to offer

> 		day == "easter day"
> 		month == "easter month"

Reviewer 2: ```suggestion
		# Easter Sunday is on a Sunday.
		"days since epoch" % 7 == 0
		# Easter Sunday occurs between 1 to 7 days inclusive after the last paschal full moon.
		# Days are counted from the end of the year to ensure consistency with leap years.
		365 - "days until year end" > ( year % 19 * 19 + ( 15 - ( 13 + year / 100 * 8 ) / 25 + year / 100 - year / 400 ) % 30 ) % 30 + 80
		365 - "days until year end" <= ( year % 19 * 19 + ( 15 - ( 13 + year / 100 * 8 ) / 25 + year / 100 - year / 400 ) % 30 ) % 30 + 80 + 7
```

Reviewer 3: Nice.

	on offer
		conversation

> 			`During a walk through the spaceport, you observe a small white church. Written on the door in large letters are the words "Stop by for a celebration of the Risen God!"`

Reviewer: "White" is maybe not the word I'd use to describe a church, especially these days. Maybe (almost definitely) it's just me finding problems that don't exist, but I think one adjective would work better anyways.

```suggestion
			`During a walk through the spaceport, you observe a small church. Written on the door in large letters are the words "Stop by for a celebration of the Risen God!"`
```

or maybe even

```suggestion
			`During a walk through the spaceport, you observe a small church tucked between a grain silo and a welding shop. Written on the door in large letters are the words "Stop by for a celebration of the Risen God!"`
```

			choice
				`	(Step in.)`
				`	(Move on.)`
					decline
			`	Inside the building, a group is singing with deafening loudness. On the stage at the front of the room, a man strums a guitar; behind him a lady softly plays the piano. A large screen displays lyrics to the current song, which you note is "Christ Our Lord has Risen Again."`
			choice
				`	(Head out.)`
					goto headout
				`	(Stay and enjoy the music.)`

> 			`	The pastor loops through several other hymns, eventually ending with an announcement of thanks to those who attended the celebration. As the group breaks up, someone hands you a ticket to a concert next week on the other side of the planet. It's too bad you won't be able to attend.`

Reviewer: I mean, you *could* attend...

Author: ~~that would require me to write another mission~~
I don't know. Could you? I envision the player as landing at the same spaceport the majority of the time, and only going elsewhere on the planet for missions (like in the Skadenga Saga)
Also, attending in one week's time would mean you have to get the date _exact_, which might be difficult to do.

Reviewer 4: Could also swap over to being a [church bulletin](https://en.m.wikipedia.org/wiki/Parish_magazine) to talk about things the player would more strictly be disinterested in and more directly related, like an upcoming parishioner potluck or auditions to join the church choir.

Reviewer: I like the bulletin idea, because it honestly makes more sense than having you be handed a ticket to a concert at a church (which is very unlikely to happen most of the time in most churches today, even at Easter). Alternatively, you could have a short easter egg mission for if you land on the planet on exactly the right day.

I guess whether you could attend it depends whether the planet has multiple spaceports or not...

				decline
			label headout

> 			`	Outside of the door, after you exit the room, someone hands you a ticket to a concert next week on the other side of the planet. It's too bad you won't be able to attend.`

Reviewer: ```suggestion
			`	On your way out of the room, someone hands you a ticket to a concert next week on the other side of the planet. It's too bad you won't be able to attend.`
```

			label backtoship
			`	With the atmosphere steeped in nostalgia for a distant city on a distant world long past, you return to your ship.`

mission "Set Easter Date"
	landing
	repeat
	non-blocking
	to offer

> 		"easter: a" != "year" * 19

Reviewer 5: Should this not also be a modulo? Otherwise, this condition is never going to be equal to the year times 19.
```suggestion
		"easter: a" != "year" % 19
```

		"easter: a" = "year" % 19
		"easter: b" = ( ( ( "year" % 100 ) - "year" ) / -100 )
		"easter: c" = "year" % 100
		"easter: d" = ( ( ( "easter: b" % 4 ) - "easter: b" ) / -4 )
		"easter: e" = "easter: b" % 4

		"easter: g" = ( ( ( ( "easter: b" * 8 + 13 ) % 25 ) - ( "easter: b" * 8 + 13 ) ) / -25 )
		"easter: h" = ( ( 19 * "easter: a" + "easter: b" - "easter: d" - "easter: g" + 15 ) % 30 )
		"easter: i" = ( ( ( "easter: c" % 4 ) - "easter: c" ) / -4 )

		"easter: k" = "easter: c" % 4
		"easter: l" = ( ( 32 + 2 * "easter: e" + 2 * "easter: i" - "easter: h" - "easter: k" ) % 7 )
		"easter: m" = ( ( ( ( "easter: a" + 11 * "easter: h" + 19 * "easter: l" ) % 433 ) - ( "easter: a" + 11 * "easter: h" + 19 * "easter: l" ) ) / -433 )
		"easter: n" = ( ( ( ( "easter: h" + "easter: l" - 7 * "easter: m" + 90 ) % 25 ) - ( "easter: h" + "easter: l" - 7 * "easter: m" + 90 ) ) / -25 )

		"easter: p" = ( ( "easter: h" + "easter: l" - 7 * "easter: m" + 33 * "easter: n" + 19 ) % 32 )

		"easter month" = "easter: n"
		"easter day" = "easter: p"
		fail

mission "Easter Celebrations"
	minor
	source
		attributes "religious"

> 		government "Deep" "Free Worlds" "Independent" "Neutral" "Pirate" "Republic" "Syndicate"

Reviewer 5: I think there are only three planets in human space that have the "religious" attribute: Harmony, New Tibet, and New Tortuga.
Harmony gets the attribute for its Amish population, who are Christians, so I think it makes sense there (although, I'm not sure how people who do not wish to use machines are planning to attend a concert on the other side of the planet in a week).
New Tibet, however, gets the attribute for its Buddhist population. As far as I know, Buddhists don't celebrate Easter.

Reviewer 6: Where does it say that there are Amish on Harmony?

Author: > The first settlers who came to Harmony were an Amish community seeking a place [...]

Later in the description it implies that a majority of the population are still Amish.

Reviewer 6: Ah right, Github search sucks (as usual)

			label backtoship
			`	With the atmosphere steeped in nostalgia for a distant city on a distant world long past, you return to your ship.`

mission "Easter Celebrations"
	minor
	source

> 		planet "Harmony" "New Tortuga"

Reviewer 7: Why such a limited list of planets?

		365 - "days until year end" <= ( year % 19 * 19 + ( 15 - ( 13 + year / 100 * 8 ) / 25 + year / 100 - year / 400 ) % 30 ) % 30 + 80 + 7
	on offer
		conversation
			`During a walk through the spaceport, you observe a small church tucked between a grain silo and a welding shop. Written on the door in large letters are the words "Stop by for a celebration of the Risen God!"`
			choice
				`	(Step in.)`
				`	(Move on.)`
					decline
			`	Inside the building, a group is singing with deafening loudness. On the stage at the front of the room, a man strums a guitar; behind him a lady softly plays the piano. A large screen displays lyrics to the current song, "Christ Our Lord has Risen Again."`
			choice
				`	(Head out.)`
					goto headout
				`	(Stay and enjoy the music.)`
			`	The pastor loops through several other hymns, eventually ending with an announcement of thanks to those who attended the celebration. As the group breaks up, you notice a bulletin board documenting several church events - a potluck on Wednesday, a notification of an outing on Saturday night, and several other items blocked by a passing pedestrian.`
				decline
			label headout

> 			`	On your way out of the room, you notice a bulletin board documenting several church events - a potluck on Wednesday, a notification of an outing on Saturday night, and several other items hidden by someone else.`

Reviewer 7: ```suggestion
			`	On your way out of the room, you notice a bulletin board documenting several church events - a potluck on Wednesday, a notification of an outing on Saturday night, and several other items blocked by a passing pedestrian.`
```
